lib/java/test-utils/src/main/java/org/enso/test/utils/ContextUtils.java

```diff
@@ -248,11 +357,118 @@ public static Set<String> allMethodsFromAny(Context ctx) {
         .collect(Collectors.toUnmodifiableSet());
   }
 
-  private static String unqualifiedName(String name) {
-    if (name.contains(".")) {
-      return name.substring(name.lastIndexOf('.') + 1);
+  private Context currentCtx() {
+    if (context == null) {
+      context = ctxBldr.build();
+    }
+    return context;
+  }
+
+  public static final class Builder {
+    private Context.Builder polyglotCtxBldr;
+    private final ByteArrayOutputStream stdout = new ByteArrayOutputStream();
+    private final ByteArrayOutputStream stderr = new ByteArrayOutputStream();
+    private boolean alwaysExecuteInContext = true;
+
+    private Builder(String... permittedLanguages) {
+      this.polyglotCtxBldr = defaultContextBuilder(permittedLanguages);
+      this.polyglotCtxBldr.out(stdout).err(stderr).logHandler(stdout);
+    }
+
+    private static Context.Builder defaultContextBuilder(String... permittedLanguages) {
+      return Context.newBuilder(permittedLanguages)
+          .allowExperimentalOptions(true)
+          .allowIO(IOAccess.ALL)
+          .allowAllAccess(true)
+          .option(RuntimeOptions.LOG_LEVEL, Level.WARNING.getName())
+          .option(RuntimeOptions.DISABLE_IR_CACHES, "true")
+          .option(RuntimeOptions.STRICT_ERRORS, "true")
+          .option(
+              RuntimeOptions.LANGUAGE_HOME_OVERRIDE,
+              Paths.get("../../distribution/component").toFile().getAbsolutePath());
+    }
+
+    /** Modifies the {@link Context.Builder}. Can be called multiple times. */
+    public Builder withModifiedContext(Function<Context.Builder, Context.Builder> modifier) {
+      polyglotCtxBldr = modifier.apply(polyglotCtxBldr);
+      return this;
+    }
+
+    /**
+     * Shortcut for {@code withModifiedContext(b -> b.option(RuntimeOptions.PROJECT_ROOT,
+     * projRoot)}.
+     */
+    public Builder withProjectRoot(Path projectRootDir) {
+      polyglotCtxBldr.option(
+          RuntimeOptions.PROJECT_ROOT, projectRootDir.toAbsolutePath().toString());
+      return this;
+    }
+
+    /**
+     * Whether the code in the tests should be executed in the polyglot context. A necessity for
+     * executing artificially created Truffle nodes. This basically ensures that executing {@link
+     * EnsoContext#get(Node)} will always return non-null value in the test.
+     *
+     * <p>Is true by default.
+     *
+     * @param b true for automatically wrapping the test code in the context. If false, the context
+     *     entering must be done manually.
+     */
+    public Builder alwaysExecuteInContext(boolean b) {
```

Comment:
I don't see a single use of this method. Can we 
```suggestion
    private Builder alwaysExecuteInContext(boolean b) {
```
for now?

Reply:
OK. Done in 342de0bbe74668be0420261096ecbc61b8aeda16